core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2250,6 +2250,25 @@ class SparkContext(config: SparkConf) extends Logging {
   }
 
   /**
+   * Kill a given task. It will be retried.
+   *
+   * @param taskId the task ID to kill
+   */
+  def killTask(taskId: Long): Unit = {
+    killTask(taskId, "cancelled")
+  }
+
+  /**
+   * Kill a given task. It will be retried.
+   *
+   * @param taskId the task ID to kill
+   * @param reason the reason for killing the task, which should be a short string
+   */
+  def killTask(taskId: Long, reason: String): Unit = {
```

Comment:
Given Mridul's points maybe killTaskAttempt is a better name?  IMO specifying "attempt" in the name makes it sound less permanent than killTask (which to me sounds like it won't be retried)